src/silo/prepare_dataset.cpp

```diff
@@ -20,28 +20,24 @@
 ) {
    SPDLOG_INFO("Pruning metadata");
 
-   std::unordered_set<uint64_t> epi_isl_ids;
+   std::unordered_set<std::string> epi_isl_set;
```

Comment:
```suggestion
   std::unordered_set<std::string> found_primary_keys;
```
I usually don't see any value in adding the type (like `set` or `map`) to a variable name.

Also this should be more general. I think we don't want to stick to "epi_isls" anymore. From my understanding, FASTA files consist of headers and the sequences. The header looks like ">identifier". We expect that the identifier is our primary key, isn't it? We should reflect this is variable naming here.